templates/common/infra/bicep/core/database/cosmos/mongo/cosmos-mongo-cluster.bicep

```diff
@@ -0,0 +1,62 @@
+metadata description = 'Azure Cosmos DB MongoDB vCore cluster'
+@maxLength(40)
+param name string
+param location string = resourceGroup().location
+param tags object = {}
+
+param administratorLogin string
+param sku string
+param storage int
+param nodeCount int
+@secure()
+param administratorLoginPassword string
+param highAvailabilityMode bool = false
+param allowAzureIPsFirewall bool = false
+param allowAllIPsFirewall bool = false
+param allowedSingleIPs array = []
+
+resource mognoCluster 'Microsoft.DocumentDB/mongoClusters@2023-03-01-preview' = {
+  name: name
+  tags: tags
+  location: location
+  properties: {
+    administratorLogin: administratorLogin
+    administratorLoginPassword: administratorLoginPassword
+    nodeGroupSpecs: [
+        {
+            diskSizeGB: storage
+            enableHa: highAvailabilityMode
+            kind: 'Shard'
+            nodeCount: nodeCount
+            sku: sku
+        }
+    ]
+  }
+
+  resource firewall_all 'firewallRules' = if (allowAllIPsFirewall) {
+    name: 'allow-all-IPs'
+    properties: {
+      startIpAddress: '0.0.0.0'
+      endIpAddress: '255.255.255.255'
+    }
+  }

```

Comment:
We get push back when we open up to all IPs like this.  Should we consider a private endpoint

Reply:
That's how it was implemented in other biceps for firewall rules.

So only these three options not sure why anyone might use this option but I followed the other biceps

Reply:
@vhvb1989  thoughts?

Reply:
Yeah, I think we should not have the `allowAll` option as part of a /core bicep module.

We don't have a way to warn folks about the security risk.  See the message from the Portal:

![image](https://github.com/Azure/azure-dev/assets/24213737/1f7dda3d-f53e-4295-91dd-f58da0931b3d)


The closest thing to this would be to add something like:

```bicep
@description('Warning: No IP addresses will be blocked and any host on the Internet can access the coordinator in this server group. It is strongly recommended to use this rule only temporarily and only on test clusters that do not contain sensitive data.')
param allowAllIPsFirewall bool = false
```

But, my vote would be for hiding this option and making it specifically a `manual-effort` to customers to open the cluster to all the world.

Reply:
Done.
Please let me know how you would like to proceed with all Ips thing. Should I remove it or not?
I added Victor's warning.

Reply:
I think this was done originally to enable local IP address to be unblocked.  But, maybe opened it up too wide.  It would be nice if we could parameterize the bicep to accept an IP to unblock and then have azd auto-populate that with the dev's IP.   @vhvb1989  thoughts?

Reply:
@jongio this bicep currently have the option to allow specific IPs only you give it to the list `allowedSingleIPs` and it will add them in the firewall rules.

IMO, it was created like this to match the deployment options provided by the Azure portal as you also have these three options available in the networking tab when you try to deploy this resource.

Reply:
> have azd auto-populate that with the dev's IP

That sounds like a new issue to me, @jongio,  maybe as a follow up.
For this PR-only scope, I think it is good to go.

Thank you for adding the description for params @john0isaac